File: Assets/ProjectGame/Lib/data/enum/GameScene/GameSceneID.js
// GameSceneID - Pure JavaScript Version
export const GameSceneID = {
    None: 0,  // デフォルト値（C#互換）
    MorningRoom: 1,  // MorningRoom
    Max: 2
};

export const GameSceneIDExtensions = {
    /**
     * Enumを数値に変換
     */
    toInt(id) {
        return Number(id);
    },

    /**
     * 数値をEnumに変換
     */
    toGameSceneID(id) {
        return id;
    },

    /**
     * 0-based indexに変換
     */
    toIndex(id) {
        return Number(id) - 1;
    },

    /**
     * すべてのIDに対して処理を実行
     */
    forID(action) {
        if (typeof action !== 'function') {
            throw new Error('action must be a function');
        }
        const start = GameSceneID.MorningRoom;
        const max = GameSceneID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameSceneID).includes(id)) {
                action(id);
            }
        }
    },

    /**
     * 条件に合うすべてのIDを返す
     */
    findAll(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const results = [];
        const start = GameSceneID.MorningRoom;
        const max = GameSceneID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameSceneID).includes(id)) {
                if (predicate(id)) {
                    results.push(id);
                }
            }
        }
        return results;
    },

    /**
     * 条件に合う最初のIDを返す（見つからなければ None）
     */
    find(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const start = GameSceneID.MorningRoom;
        const max = GameSceneID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameSceneID).includes(id)) {
                if (predicate(id)) {
                    return id;
                }
            }
        }
        return GameSceneID.None;
    }
};

File: Assets/ProjectGame/Lib/data/enum/GameScene/GameSceneID.py
from enum import IntEnum
from typing import Callable, List, Any


class GameSceneID(IntEnum):
    NONE = 0  # デフォルト値（C#のNoneに相当）
    MorningRoom = 1  # MorningRoom
    MAX = 2

def to_int(id: "GameSceneID") -> int:
    return int(id)


def to_GameSceneID(id: int) -> "GameSceneID":
    return GameSceneID(id)


def to_index(id: "GameSceneID") -> int:
    return int(id) - 1


def for_id(action: Callable[["GameSceneID"], None]):
    if action is None:
        raise ValueError("action cannot be None")
    start = GameSceneID.MorningRoom.value
    for i in range(start, GameSceneID.MAX.value):
        try:
            value = GameSceneID(i)
            action(value)
        except ValueError:
            continue  # 未定義の値はスキップ


def find_all(predicate: Callable[["GameSceneID"], bool]) -> List["GameSceneID"]:
    if predicate is None:
        raise ValueError("predicate cannot be None")
    results: List["GameSceneID"] = []
    start = GameSceneID.MorningRoom.value
    for i in range(start, GameSceneID.MAX.value):
        try:
            value = GameSceneID(i)
            if predicate(value):
                results.append(value)
        except ValueError:
            continue
    return results


def find(predicate: Callable[["GameSceneID"], bool]) -> "GameSceneID":
    if predicate is None:
        raise ValueError("predicate cannot be None")
    start = GameSceneID.MorningRoom.value
    for i in range(start, GameSceneID.MAX.value):
        try:
            value = GameSceneID(i)
            if predicate(value):
                return value
        except ValueError:
            continue
    return GameSceneID.NONE
